src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,56 @@
+package duke;
+
+import duke.command.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/** A class that deals with making sense of the user input, and translating it to a command for the bot to execute. */
+public class Parser {
+
+    private static final ArrayList<String> ADD_COMMANDS = new ArrayList<>(List.of("todo", "deadline", "event"));
+    private static final ArrayList<String> DELETE_COMMANDS = new ArrayList<>(List.of("delete", "remove"));
+    private static final ArrayList<String> EDIT_COMMANDS = new ArrayList<>(List.of("mark", "unmark"));
+    private static final ArrayList<String> VIEW_COMMANDS = new ArrayList<>(List.of("list", "find"));
+    private static final ArrayList<String> EXIT_COMMANDS = new ArrayList<>(List.of("bye"));
+
+    public Parser() {
+        //
+    }
+
+    /**
+     * Parses the user input into a command for the bot.
+     *
+     * @param strToParse The user input.
+     * @return A Command that can be executed by the bot.
+     * @throws DukeException If the user gives an invalid input that the bot cannot understand.
+     */
+    public static Command parse(String strToParse) throws DukeException {
+        String[] splitTask = strToParse.split(" ", 2);
+        String keyword = splitTask[0];
+        if (!ADD_COMMANDS.contains(keyword) && !EDIT_COMMANDS.contains(keyword) && !VIEW_COMMANDS.contains(keyword)
```

Comment:
Perhaps using a switch statement here could be better so that it is more readable